src/cli.ts

```diff
@@ -182,3 +189,44 @@ function prepareCliMessage(
 
 	return spaced ? `\n${preparedMessage}\n` : preparedMessage;
 }
+
+export function printEnvInfo(): void {
+	cliLog(`
+		System:
+			Platform: ${os.platform()}
+			Arch: ${os.arch()}
+			Version: ${os.version()}
+			CPU: (${os.cpus().length}) ${os.arch()} ${os.cpus()[0]?.model}
+			Memory: ${Math.round(os.totalmem() / 1024 / 1024 / 1024)} GB
+			Shell: ${process.env['SHELL']?.toString() ?? 'Unknown'}
+		Binaries:
+			Node: ${process.versions.node}
+			Yarn: ${getBinaryVersion('yarn')}
+			npm: ${getBinaryVersion('npm')}
+			pnpm: ${getBinaryVersion('pnpm')}
```

Comment:
On Windows, these need to use `{name}.cmd`. You could use the utility in `utils/getSpawnCommand.ts` to do this.